packages/core/src/core/rename.ts

```diff
@@ -87,27 +94,29 @@ const getRenameElementChanges = async (
   sourceElemId: ElemID,
   targetElemId: ElemID,
   sourceElements: InstanceElement[],
-): Promise<DetailedChange[]> => {
+): Promise<DetailedChangeWithBaseChange[]> => {
+  const baseRemoveChange = toChange({ before: sourceElements[0] })
```

Comment:
instead of building the change explicitly
```suggestion
  const removeChanges = getDetailedChanges(toChange({ before: sourceElements[0] }))
```

applies to several places in this PR (in other files as well)